### PR TITLE
Fix the following bugs

### PR DESCRIPTION
**Pull request by HakkyuKim**
## Current bugs

Altering different build modes doesn't generate expected tpks

- debug -> release : output tpk contains debug binaries because `{OUTPUT_DIR}/flutter_assets` directory still contains them when `DotnetTpk.build()` or `NativeTpk.build()` is called.
- release -> debug: output tpk doesn't contain debug binaries because `TizenAssetBundle.build()` is not reinvoked due to caching. Debug binaries must be specified in `TizenAssetBundle.outputs`, see `DebugAndroidApplication` in `android.dart`.
- x86 -> arm, arm -> x86: arch string is not provided to `Enviroment` in `TizenBuilder.buildTpk()`.

## Changes

- create DebugTizenAssetBundle and ReleaseTizenAssetBundle.
Both targets must inherit AssetBundle. (See Debug
AndroidApplication and AotAndroidAssetBundle in android.dart)

- Package tpk after FlutterBuildSystem syncs output directory

- pass arch string to Environment instantiation to
force different hash values in .dart_tool for different
arch types. (See kIosArches in common.dart)

### CODE REVIEW

**Comment by wiertel** [comment]
how does it relate to #55? if they are independent then ok

**Comment by HakkyuKim** [comment]
> how does it relate to #55? if they are independent then ok

This pr is a pre required implementation for #55. #55 was an attempt to implement the feature as a single pr, but it is a bit complex so I've decided to break it down into smaller steps. I'll add necessary labels or notes in #55 if necessary.

**Comment by wiertel** [comment]
ok. Thanks for the info. I suspected that this might build upon #55 - i.e. #61 depends on #55, as this is newer.
I think no more notes in #55 are needed as it will be most probably you who will merge the PRs, and there is no risk that they will be merged in wrong order.

File: lib/tizen_build_target.dart [hunk 1]
```diff
@@ -27,21 +27,60 @@ import 'tizen_tpk.dart';
...
-    await super.build(environment);
-  }
+  List<Target> get dependencies => <Target>[
+        ...super.dependencies,
+        TizenPlugins(project, buildInfo),
+      ];
+}
+
+/// Source: [DebugAndroidApplication] in `android.dart`
+class DebugTizenAssetBundle extends TizenAssetBundle {
```
[the remark this answers is not in the document]
Comment (HakkyuKim) [reply lib/tizen_build_target.dart:46]: Will do the same for `ReleaseTizenAssetBundle` -> `ReleaseTizenApplication`

[the remark this answers is not in the document]
Comment (swift-kim) [reply lib/tizen_build_target.dart:46]: Yep

File: lib/tizen_build_target.dart [hunk 2]
```diff
@@ -27,21 +27,60 @@ import 'tizen_tpk.dart';
...
-    final Directory flutterAssetsDir =
-        environment.outputDir.childDirectory('flutter_assets');
-    if (flutterAssetsDir.existsSync()) {
-      flutterAssetsDir.deleteSync(recursive: true);
-    }
-    await super.build(environment);
-  }
+  List<Target> get dependencies => <Target>[
+        ...super.dependencies,
+        TizenPlugins(project, buildInfo),
```
[the remark this answers is not in the document]
Comment (HakkyuKim) [reply lib/tizen_build_target.dart:41]: The quick alternative would be pulling down the field to `DebugTizenApplication` and `ReleaseTizenApplication` like the following:

```dart
class DebugTizenApplication extends TizenAssetBundle{

@override
List<Target> get dependencies => <Target>[
        ...super.dependencies,
        if (TizenProject.fromFlutter(project).isDotnet)
          TizenPlugins(project, buildInfo),
      ];
}

class ReleaseTizenApplication extends TizenAssetBundle{

@override
List<Target> get dependencies => <Target>[
        ...super.dependencies,
        TizenAotElf(buildInfo.targetArchs),
        if (TizenProject.fromFlutter(project).isDotnet)
          TizenPlugins(project, buildInfo),
      ];
}
```

`isDotnet` is added to depend `TizenPlugins` only for Dotnet projects. It looks more clear if we don't mind the repetition(?).

[the remark this answers is not in the document]
Comment (swift-kim) [reply lib/tizen_build_target.dart:41]: That looks okay.

**Comment by swift-kim** [comment]
How about this: https://github.com/swift-kim/flutter-tizen/commit/0826fb76088fa75fd4f6b5d3a156884b1fc90fa8

Otherwise you may have to remove the properties `project` and `buildInfo` from `TizenAssetBundle` as they have basically nothing to do with the name "TizenAssetBundle".

**Comment by HakkyuKim** [comment]
> How about this: [swift-kim@0826fb7](https://github.com/swift-kim/flutter-tizen/commit/0826fb76088fa75fd4f6b5d3a156884b1fc90fa8)
> 
> Otherwise you may have to remove the properties `project` and `buildInfo` from `TizenAssetBundle` as they have basically nothing to do with the name "TizenAssetBundle".

How about copying the `AndroidAssetBundle` code to `TizenAssetBundle` after removing the inheritance, and pulling down properties `project` and `buildInfo` from `TizenAssetBundle`? (Similar to `IosAssetBundle` in `ios.dart`)
HakkyuKim@221f315cce42c5c5099c30f1e0eed2a4fa1060ca

File: lib/tizen_builder.dart [hunk 3]
```diff
@@ -137,6 +135,12 @@ class TizenBuilder {
         throwToolExit('The build failed.');
       }
 
+      if (tizenProject.isDotnet) {
+        await DotnetTpk(project, tizenBuildInfo).build(environment);
+      } else {
+        await NativeTpk(project, tizenBuildInfo).build(environment);
```
Comment (swift-kim) [inline lib/tizen_builder.dart:141]: This part may be improved later.

**Review by swift-kim — APPROVED** [review]

**Pull request merged**